engineering-health-check/cxInsight_9_0.ps1

```diff
@@ -70,47 +80,150 @@ function getOAuth2Token() {
+function getScanOdata {
+    param (
+        $outputFile
+    )
+    Write-Verbose "Retrieving scan data"
+
+    $Url = "${cx_sast_server}/cxwebinterface/odata/v1/Scans?`$select=Id,ProjectId,OwningTeamId,TeamName,ProductVersion,EngineServerId,Origin,PresetName,ScanRequestedOn,QueuedOn,EngineStartedOn,EngineFinishedOn,ScanCompletedOn,ScanDuration,FileCount,LOC,FailedLOC,TotalVulnerabilities,High,Medium,Low,Info,IsIncremental,IsLocked,IsPublic&`$expand=ScannedLanguages(`$select=LanguageName)&`$filter=ScanRequestedOn%20gt%20${start_date}Z%20and%20ScanRequestedOn%20lt%20${end_date}z"
     $headers = @{
         Authorization = $token
     }
     try {
         if ($bypassProxy) {
-            $response = Invoke-RestMethod -noProxy -uri "$Url" -method get -headers $headers -OutFile "data.txt"
+            $response = Invoke-RestMethod -noProxy -uri "$Url" -method get -headers $headers -OutFile $outputFile
         }
         else {
-            $response = Invoke-RestMethod -uri "$Url" -method get -headers $headers -OutFile "data.txt"
+            $response = Invoke-RestMethod -uri "$Url" -method get -headers $headers -OutFile $outputFile
         }
-        return $response
     }
     catch {
         Write-Host "Exception:" $_
         Write-Host "StatusCode:" $_.Exception.Response.StatusCode.value__
         Write-Host "StatusDescription:" $_.Exception.Response.StatusDescription
-		Write-Host $Url
-		Write-Host "An Error has prevented this script from collecting Odata."
+	Write-Host $Url
+	Write-Host "An error has prevented this script from collecting scan Odata."
+        exit(-1)
+    }
+}
+
+function getResultOData {
+    param (
+        $outputFile
+    )
+    Write-Verbose "Retrieving result data"
+
+    $Url = "${cx_sast_server}/cxwebinterface/odata/v1/Projects?`$select=Id"
+    $headers = @{
+        Authorization = $token
+    }
+    try {
+        if ($bypassProxy) {
+            $response = Invoke-RestMethod -noProxy -uri "$Url" -method get -headers $headers
+        }
+        else {
+            $response = Invoke-RestMethod -uri "$Url" -method get -headers $headers
+        }
+
+        $states = @{
+            "0" = "To Verify"
+            "1" = "Not Explotable"
+            "2" = "Confirmed"
+            "3" = "Urgent"
+            "4" = "Proposed Not Exploitable"
+        }
+        $projects = @{}
+        $response | Select-Object -ExpandProperty Value | ForEach-Object {
+            $projectId = "$($_.Id)"
+            Write-Verbose "Retrieving result data for project ${projectId}"
+
+            $Url = "${cx_sast_server}/cxwebinterface/odata/v1/Projects(${projectId})?`$select=Id&`$expand=LastScan(`$select=Id;`$expand=Results(`$select=Id,ResultId,StateId))"
+            if ($bypassProxy) {
+                $response = Invoke-RestMethod -noProxy -uri "$Url" -method get -headers $headers
+            }
+            else {
+                $response = Invoke-RestMethod -uri "$Url" -method get -headers $headers
+            }

```

Comment:
Similar variations of the OData API call code is repeated about 4 times.  There should probably be a function that does the common part.